src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/CaseDetails.java

```diff
@@ -29,6 +36,34 @@
     private String caseTypeId;
     @JsonProperty("case_data")
     private Map<String, JsonNode> data;
+    @JsonProperty("security_classification")
+    private SecurityClassification securityClassification;
+    @JsonProperty("data_classification")
+    @ApiModelProperty("Same structure as `case_data` with classification (`PUBLIC`, `PRIVATE`, `RESTRICTED`) "
+        + "as field's value.")
+    private Map<String, JsonNode> dataClassification;
+
+    public Optional<String> findChangeOrganisationRequestFieldName() {
+        Optional<JsonNode> first = getData().values().stream()
+            .map(node -> node.findParents(ORGANISATION_TO_ADD))
+            .flatMap(List::stream)
+            .map(node -> node.findParents(ORGANISATION_TO_REMOVE))
+            .flatMap(List::stream)
+            .map(node -> node.findParents(ORGANISATION_CASE_ROLE_ID))
+            .flatMap(List::stream)
+            .map(node -> node.findParents(ORGANISATION_REQUEST_TIMESTAMP))
+            .flatMap(List::stream)
+            .findFirst();
+
+        if (first.isPresent()) {
+            for (Map.Entry<String, JsonNode> entry : getData().entrySet()) {
+                if (entry.getValue().equals(first.get())) {
```

Comment:
we could do Optional result = getData().entrySet().stream().filter ( entry -> entry.getValue.equals( first.get()  )